### ISSUE TITLE
Add search type picker to the Find It page

### ISSUE THREAD

**Issue by eric-gilbertson** [RocketMan/zookeeper#174]
Add search type picker to the Find It page. The functionality of Find It would be increased by allowing the user the ability to filter the results by specifying the search type, eg, all, review, playlist, album...

**Comment by RocketMan** [comment]
You were getting a lot of false positives because of the stopword issue.  As a result, the results list was much larger than it should have been.

Now that the stopwords have been adjusted, let's try it for a while and see if the results are not more amenable.

Conceptually, I resist adding a filter option to Find It!.  Find It! is supposed to be like modern google or OS X searches, where It Just Works.  Clean and Simple.  If it is not working, then we fix what's necessary on our end rather than making it the user's problem.

**Comment by eric-gilbertson** [comment]
I agree with the just works ethos but adding a filter OPTION makes the page more effective in a way that is consistent with it. Even with the stop word change, it is possible to get a lot of unwanted results, eg when DJs  searching their playlists for say Kinks plays there is no point in displaying hits in the four other categories and all the extra text makes it difficult to zero in on the playlist hits. Note that the default value of the filter would be "All" so that the current behavior is preserved by default. In addition it would be helpful to an "exact match" option in order to make this option explicit.

**Comment by RocketMan** [comment]
First a filter, then an exact match box.  It's 1995 all over again.  We are not going down this path.

Sure, we kept Classic Search because it still has value, but we don't want to remake Find It! in its image.

Engineers think differently to non-engineers.  Engineers love tweaking filters and ticking boxes, non-engineers just want it to work.  Think about people like Byrd.  He doesn't go to zookeeper to play around with options.

Moreover, I disagree that Find It! returns a lot of unwanted results, because it usually returns only 5 per category, with the option to drill in for more.  Even if you search something completely generic like 'jazz', you get only one page of results.

**Comment by eric-gilbertson** [comment]
This ticket has nothing to do with engineering tweaks. Instead it has to do with usability and I would say that based upon user acceptance, my suggestions  eg track time stamping, first/last name presentation and style updates have been well  supported and made ZK a much more usable app. An example of an engineering tinkering would be to clone the code and rewrite the front end.

**Comment by RocketMan** [comment]
I think you misread me.  Not tweaking, **thinking**.  Engineers don't *think* like normal users.  Engineers love **user interfaces** that have heaps of options, because their thoughts are ordered, and it allows them laser-like focus to their goal.  That's what you're doing here.  Normal users are quite a different matter.

To the point at hand, Find It! is not so much a search function as it is an exploration feature.  It's about serendipity.  You might be looking for playlists, but the discovery of an unexpected music review in the process might change your life, or at least your show.

I value your contributions to zookeeper and want them to continue.  All ideas you have brought forward thus far have been accepted in one form or another.  However, I am not prepared to accept additional UI elements in Find It!

**Comment by RocketMan** [comment]
However, I will accept something like this, where the filter is integrated in the control (source Amazon):

![Screenshot from 2020-08-03 12-50-21](https://user-images.githubusercontent.com/6860356/89179819-6bdaee80-d588-11ea-9e3b-bf9f54a0254c.png)

The element at the left is a drop list filter.  The search icon on the right is a nice touch, and I think would help us as well.

My principal concern is having multiple controls on the page.  If you can integrate the filter and add a search icon all within the one search element, like Amazon, it's a definite winner.

**Comment by eric-gilbertson** [comment]
Yes, that is a nice way to integrate the filter function into the Find It page. In the meantime another approach that struck me was to add a "Play List" filter option to the classic search page. Both seem to be worthwhile additions but I am inclined to do the  latter first thought since it is more straightforward. Let me know what you think.

**Comment by RocketMan** [comment]
Thanks Eric, either is fine by me, I'll leave it to your inclination and availability.

I'll be happy to assist finessing the APIs that drive either search, just let me know.
